core/trino-main/src/main/java/io/trino/connector/ConnectorManager.java

```diff
@@ -337,12 +337,14 @@ private synchronized void removeConnectorInternal(CatalogName catalogName)
         MaterializedConnector materializedConnector = connectors.remove(catalogName);
         if (materializedConnector != null) {
             Connector connector = materializedConnector.getConnector();
-            try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(connector.getClass().getClassLoader())) {
-                connector.shutdown();
-            }
-            catch (Throwable t) {
-                log.error(t, "Error shutting down connector: %s", catalogName);
-            }
+            withClassLoaderOf(connector.getClass(), () -> {
```

Comment:
This is worse than the original. Let’s revert this change and don’t deprecate the constructor version.